tokio-codec/src/lines_codec.rs

```diff
@@ -49,23 +48,28 @@ impl LinesCodec {
     /// that holds the line currently being read is unbounded. An attacker could
     /// exploit this unbounded buffer by sending an unbounded amount of input
     /// without any `\n` characters, causing unbounded memory consumption.
-    pub fn set_decode_max_line_length(&mut self, limit: usize) -> &mut Self {
-        self.max_length = Some(limit - 1);
-        self
+    pub fn with_max_length(limit: usize) -> Self {
+        LinesCodec {
+            max_length: Some(limit - 1),
+            ..LinesCodec::new()
```

Comment:
Whoa, this syntax is wild.  Do you happen to have a link to the Rust docs for this?

Reply:
@tobz See https://doc.rust-lang.org/book/second-edition/ch05-01-defining-structs.html#creating-instances-from-other-instances-with-struct-update-syntax :)

It's useful for when you have a constructor that sets some subset of a struct's fields, but uses default values for the remainder, and you want to keep the default values in this constructor in sync with the constructor that sets everything to the default values. For example,  I often see this used with the `Default` trait, like this:
```rust
Foo {
    bar: 1,
    ..Foo::default()
}
```

It's also handy when you have builders and so on, like this:
```rust
struct FooBuilder {
    bar: usize,
    baz: String,
    quux: bool,
}

impl FooBuilder {
    fn new() -> Self {
        FooBuilder {
            bar: 0,
            baz: String::new(),
            quux: false,
        }
    }

    fn new_with_quux() -> Self {
        FooBuilder {
            quux: true,
            ..Self::new() 
        }
    }

    fn set_bar(self, bar: usize) -> Self {
        FooBuilder {
            bar,
            ..self
        }
    }

    // ... and so on
}
```